candy-machine/program/src/lib.rs

```diff
@@ -489,6 +431,81 @@ pub mod candy_machine {
         Ok(())
     }
 
+    pub fn set_collection_during_mint(ctx: Context<SetCollectionDuringMint>) -> ProgramResult {
+        let ixs = &ctx.accounts.instructions;
+        let previous_instruction = get_instruction_relative(-1, ixs)?;
+        if &previous_instruction.program_id != &candy_machine::id() {
+            msg!(
+                "Transaction had ix with program id {}",
+                &previous_instruction.program_id
+            );
+            return Err(ErrorCode::SuspiciousTransaction.into());
+        }
+
+        if &previous_instruction.data[0..8] != [211, 57, 6, 167, 15, 219, 35, 251] {
+            msg!(
+                "Transaction had ix with data {:?}",
+                &previous_instruction.data[0..8]
+            );
+            return Err(ErrorCode::SuspiciousTransaction.into());
+        }
+
+        let mint_ix_accounts = previous_instruction.accounts;
+
+        let mint_ix_cm = mint_ix_accounts[0].pubkey;
+        let mint_ix_metadata = mint_ix_accounts[4].pubkey;
+        let candy_key = ctx.accounts.candy_machine.key();
+        
+        if &mint_ix_cm != &candy_key {
+            msg!("Candy Machine with pubkey {} does not match the mint ix Candy Machine with pubkey {}", mint_ix_cm, candy_key);
+            return Err(ErrorCode::SuspiciousTransaction.into());
+        }
+        if mint_ix_metadata != ctx.accounts.metadata.key() {
+            msg!(
+                "Metadata with pubkey {} does not match the mint ix metadata with pubkey {}",
+                mint_ix_metadata,
+                ctx.accounts.metadata.key()
+            );
+            return Err(ErrorCode::SuspiciousTransaction.into());
+        }
+
+        let collection_pda = &ctx.accounts.collection_pda;
+        let collection_mint = ctx.accounts.collection_mint.to_account_info();
+        if &collection_pda.mint != &collection_mint.key() {
+            return Err(ErrorCode::MismatchedCollectionMint.into());
+        }
+        let seeds = [b"collection".as_ref(), candy_key.as_ref()];
+        let bump = assert_derivation(&crate::id(), &collection_pda.to_account_info(), &seeds)?;
```

Comment:
@austbot should this be crate or candy_machine? You had it using the crate from the original collections change